Project/Src/Gui/TextAreaDragDropHandler.cs

```diff
@@ -93,24 +96,35 @@ protected void OnDragDrop(object sender, DragEventArgs e)
                     var offset = textArea.Caret.Offset;
                     if (textArea.IsReadOnly(offset))
                         return;
-                    if (e.Data.GetDataPresent(typeof(DefaultSelection)))
+
+                    try
                     {
-                        var sel = (ISelection)e.Data.GetData(typeof(DefaultSelection));
-                        if (sel.ContainsPosition(textArea.Caret.Position))
-                            return;
-                        if (GetDragDropEffect(e) == DragDropEffects.Move)
+                        if (e.Data.GetDataPresent(typeof(DefaultSelection)))
                         {
-                            if (SelectionManager.SelectionIsReadOnly(textArea.Document, sel))
+                            var sel = (ISelection)e.Data.GetData("ICSharpCode.TextEditor.Document.DefaultSelection");
+                            if (sel.ContainsPosition(textArea.Caret.Position))
                                 return;
-                            var len = sel.Length;
-                            textArea.Document.Remove(sel.Offset, len);
-                            if (sel.Offset < offset)
-                                offset -= len;
+                            if (GetDragDropEffect(e) == DragDropEffects.Move)
+                            {
+                                if (SelectionManager.SelectionIsReadOnly(textArea.Document, sel))
+                                    return;
+                                var len = sel.Length;
+                                textArea.Document.Remove(sel.Offset, len);
+                                if (sel.Offset < offset)
+                                    offset -= len;
+                            }
                         }
                     }
-
+                    catch (System.InvalidCastException)
+                    {
+                        // If GetDataPresent(typeof(DefaultSelection)) threw this exception, then it's an interprocess DefaultSelection COM object that is not serializable!
+                        // In general, GetDataPresent(typeof(...)) throws InvalidCastException for drags and drops from other GitExt processes (maybe we need to make the data objects [Serializable]?)
+                        // We can get around this exception by doing GetDataPresent(String s) [using the string of the type name seems to work fine!]
+                        // Since it is interprocess data, just get the string data from it - special handling logic in try {} is only valid for selections within the current process's text editor!
+                    }
+                    
                     textArea.SelectionManager.ClearSelection();
-                    InsertString(offset, (string)e.Data.GetData(typeof(string)));
+                    InsertString(offset, (string)e.Data.GetData("System.String"));
```

Comment:
Why?

Reply:
I found that when you have a text drag-drop from a different Git Extensions process, if you call e.Data.GetData(typeof(string)), you will hit this System.InvalidCastException, but when doing e.Data.GetData("System.String"), you do not get this exception.